TAScheduler/UnitTests/tests.py

```diff
@@ -46,38 +52,476 @@ class TestAdminCreateSection(TestCase):  # Alec
+        )
+        temp_lab = Lab.objects.create(
+            section=self.hold_sec,
+            ta=None
+        )
+        self.tempLab = LabObj(temp_lab)
+        temp = User.objects.create(
+            email_address='admin@example.com',
+            password='adminpassword',
+            first_name='Admin',
+            last_name='User',
+            home_address='123 Admin St',
+            phone_number='1234567890'
+        )
+        hold_admin = Administrator(user=temp)
+        hold_admin.save()
+        self.admin = AdminObj(hold_admin)
+
+    def test_successful_delete(self):
+        self.admin.removeSection(self.tempLab)
+        self.assertNotIn(User.objects, self.hold_sec, "Did not remove section from the database")
+
+    def test_delete_null_user(self):
+        Section.delete(self.hold_sec)
+        with self.assertRaises(RuntimeError, msg="Tried to delete a non-existent section"):
+            self.admin.removeSection(self.tempLab)
+
+    def test_delete_non_user(self):
+        with self.assertRaises(TypeError, msg="Tried to delete not a section"):
+            self.admin.removeSection(11)
 
 
 class TestAdminEditCourse(TestCase):  # Kevin
-    pass
+    tempCourse = None
+    admin = None
+    new_info = None
+
+    def setUp(self):
+        hold_course = Course.objects.create(
+            course_id=101,
+            semester='Fall 2023',
+            name='Introduction to Testing',
+            description='A course about writing tests in Django.',
+            num_of_sections=3,
+            modality='Online',
+            credits=4
+        )
+        self.tempCourse = CourseObj(hold_course)
+        hold_user = User(
+            email_address='admin@example.com',
+            password='adminpassword',
+            first_name='Admin',
+            last_name='User',
+            home_address='123 Admin St',
+            phone_number='1234567890'
+        )
+        hold_user.save()
+        hold_admin = Administrator(user=hold_user)
+        hold_admin.save()
+        self.admin = AdminObj(hold_admin)
+        self.new_info = {"course id", 103,
+                         "semester", "Spring 2024",
+                         "name", "Intro to Units",
+                         "description", "Unit testing at its finest",
+                         "num of sections", 4,
+                         "modality", "",
+                         "credits", 3}
+
+    def test_bad_course(self):
+        with self.assertRaises(TypeError, msg='Course that was passed is not a valid course'):
+            self.admin.editCourse(11, self.new_info)
+
+    def test_bad_info(self):
+        with self.assertRaises(TypeError, msg='Improper input entered for editing course'):
+            self.admin.editCourse(self.tempCourse, 11)
+
+    def test_success(self):
+        self.admin.editCourse(self.tempCourse, self.new_info)
+        self.assertEqual(self.new_info["description"], Course.objects.get(course_id=103))
+
```

Comment:
good call

Reply:
If the course does not exist in the database then this test does not apply and the bad course test applies.

Reply:
I don't think that's necessarily true: looking up the definition of a type error, which is what bad course test does "A TypeError may be thrown when: an operand or argument passed to a function is incompatible with the type expected by that operator or function." If we pass in a course in the database, then that is not an incompatible type - this method is designed to take in courses as an argument, which is what my test would send in.

It may be some sort of ValueError ("Python's ValueError occurs when a function is called with the proper argument type but with the wrong value.") 

I strongly recommend simply adding 1 test to test what I was suggesting.